biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceUtils.java

```diff
@@ -546,6 +546,8 @@ public static VersionedClause toVersionClause(Resource resource, String mask) {
 		if (resource.getCapabilities(WORKSPACE_NAMESPACE).isEmpty()) {
 			Macro macro = new Macro(new Processor());
 			Version version = getVersion(idCap);
+			if (version == null)
+				version = Version.emptyVersion;
```

Comment:
I think I agree with BJ. The right solutions is to validate the resources before they are used like this. I think we need a method in ResourceUtils that validates all the capabilities and requirements in the resource, namespace specific?